### DISCUSSION TITLE
Is StorageProvider API meant for consumption?

### DISCUSSION THREAD

**Discussion by jhchee** [jobrunr/jobrunr#724] [Q&A] [answered]
I notice StorageProvider has provided a way to get job list updated before an instant.

```java
List<Job> getJobs(StateName state, Instant updatedBefore, PageRequest pageRequest);
```

Is the StorageProvider API meant for consumption (it is with public access 👀)? Will the API evolve radically over times?

**Comment by rdehuyss** [comment, marked as answer]
No, the StorageProvider is a public API and breaking backwards compatibility will result in a new major version of JobRunr.
